perfkitbenchmarker/publisher.py

```diff
@@ -554,7 +554,11 @@ def AddSamples(self, samples, benchmark, benchmark_spec):
       sample['product_name'] = FLAGS.product_name
       sample['official'] = FLAGS.official
       sample['owner'] = FLAGS.owner
-      sample['timestamp'] = time.time()
+      if 'timestamp' in sample['metadata']:
```

Comment:
We could, but it would complicate testing, because a lot of the benchmark tests compare the results of the benchmark against a static list of Sample objects. It wouldn't be too hard to mock the timestamp, but it would mean that all tests have to care about the time, even ones where the benchmark just uses the default time.time() option.

On the other hand, a timestamp feels like a very natural part of a sample to me. What do you think?